core/chains/evm/client/node.go

```diff
@@ -1002,6 +1001,11 @@ func (n *node) makeLiveQueryCtx(parentCtx context.Context) (ctx context.Context,
 		return
 	}
 	cancelCh := n.chStopInFlight
+	ws = n.ws
+	if n.http != nil {
+		cp := *n.http
+		http = &cp
+	}
```

Comment:
Return _copies_ of these `rawclient`s, so that the callers don't race to use them after the lock is released.